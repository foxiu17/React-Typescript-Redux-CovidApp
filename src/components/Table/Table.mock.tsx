import { IRow } from "../../Interfaces/TableInterfaces";

export const countryArr: Array<IRow> = [
  {
    Country: "Poland",
    CountryCode: "PL",
    Province: "",
    City: "",
    CityCode: "",
    Lat: "51.92",
    Lon: "19.15",
    Confirmed: 6674,
    Deaths: 232,
    Recovered: 439,
    Active: 6003,
    Date: "2020-04-13T00:00:00Z",
    LocationID: "PL"
  },
  {
    Country: "Poland",
    CountryCode: "PL",
    Province: "",
    City: "",
    CityCode: "",
    Lat: "51.92",
    Lon: "19.15",
    Confirmed: 6674,
    Deaths: 232,
    Recovered: 439,
    Active: 5777,
    Date: "2020-04-12T00:00:00Z",
    LocationID: "PL"
  }
];