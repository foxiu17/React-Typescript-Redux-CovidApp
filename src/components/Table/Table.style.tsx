import styled from "styled-components";

import {
  TableWrapper as commonTable,
  TableBody as commonTableBody,
  TableCell as commonTableCell,
  TableContainer as commonTableContainer,
  TableHead as commonTableHead,
  TableRow as commonTableRow
} from '../../Assets/Common/Tables.style';
import { Paper as commonPaper } from '../../Assets/Common/Containers.style';

export const TableWrapper = styled(commonTable)``;
export const TableBody = styled(commonTableBody)``;
export const TableCell = styled(commonTableCell)``;
export const TableContainer = styled(commonTableContainer)``;
export const TableHead = styled(commonTableHead)``;
export const TableRow = styled(commonTableRow)``;
export const Paper = styled(commonPaper)``;