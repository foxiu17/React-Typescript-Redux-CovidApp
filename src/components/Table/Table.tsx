import React from "react";
import moment from 'moment';

import { ITable, IRow } from "../../Interfaces/TableInterfaces";

import {
  Paper,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableWrapper
} from './Table.style';

const Table = ({ data }: ITable): JSX.Element => {

  console.log("TABELA: ", data);
  return (
    <Paper elevation={3}>
      <TableWrapper size="small" aria-label="a dense table">
        <TableHead>
          <TableRow>
            <TableCell>Date</TableCell>
            <TableCell align="right">Confirmed</TableCell>
            <TableCell align="right">Deaths</TableCell>
            <TableCell align="right">Recovered</TableCell>
            <TableCell align="right">Active</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {data.map((row: IRow, index: number) => (
            <TableRow key={index}>
              <TableCell component="th" scope="row">
                {moment(row.Date).format('DD.MM.YYYY')}
              </TableCell>
              <TableCell align="right">{row.Confirmed}</TableCell>
              <TableCell align="right">{row.Deaths}</TableCell>
              <TableCell align="right">{row.Recovered}</TableCell>
              <TableCell align="right">{row.Active}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </TableWrapper>
    </Paper>
  );
};

export default Table;