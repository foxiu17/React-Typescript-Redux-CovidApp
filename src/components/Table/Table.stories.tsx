import React from "react";

import Table from "./Table";

import { countryArr } from "./Table.mock";

export default {
  component: Table,
  title: "Table"
};

// @ts-ignore
export const withData = () => <Table data={countryArr} />;