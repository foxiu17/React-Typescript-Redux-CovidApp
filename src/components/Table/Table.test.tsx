import React from "react";
import {shallow, mount, ReactWrapper, ShallowWrapper} from "enzyme";

import { countryArr } from "./Table.mock";
import Table from "./Table";

describe('Test Table component', () => {
  describe('Render tests', () => {
    test('Render component without crashing', () => {
      // @ts-ignore
      const component: ShallowWrapper = shallow(<Table data={countryArr}/>);
      expect(component.exists()).toBe(true);
      expect(component).toMatchSnapshot();
    })
  })
})