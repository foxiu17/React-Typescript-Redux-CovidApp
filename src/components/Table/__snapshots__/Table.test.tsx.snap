// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table component Render tests Render component without crashing 1`] = `ShallowWrapper {}`;
